src/supplemental/mqtt/mqtt_msg.c

```diff
@@ -508,7 +509,7 @@ nni_mqtt_msg_set_unsubscribe_topics(
 	for (size_t i = 0; i < topic_count; i++) {
 		nni_mqtt_topic_array_set(
 		    proto_data->payload.unsubscribe.topic_arr, i,
-		    (const char *) topics[i].buf);
+		    (const char *) topics[i].buf, topics[i].length);
```

Comment:
previous bug?

Reply:
I think so. Length is an unused arg in unsub before, now it is fixed and works just like sub.